app/controllers/events_controller.rb

```diff
@@ -0,0 +1,45 @@
+class EventsController < ApplicationController
+  before_action :logged_in?, except: %i[show index]
+
+  def index
+    @events = Event.all
+  end
```

Comment:
Your index action is expected to:
* [ ] lists past events, i.e. `@past_events = Event.previous_events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/past_upcoming_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
* [ ] lists ucomping events, i.e. `@upcoming_events = Event.upcoming_events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/past_upcoming_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))